libs/ledger-live-common/src/families/tezos/signOperation.ts

```diff
@@ -32,53 +33,101 @@ export const signOperation: SignOperationFnSignature<Transaction> = ({
             false,
             DerivationType.ED25519,
           );
+
           tezos.setProvider({ signer: ledgerSigner });
 
-          // Disable the broadcast because we want to do it in a second phase (broadcast hook)
-          // Use a dummy transaction hash, we don't care about this check
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          tezos.contract.context.injector.inject = async () =>
-            "op4WsnE6gvDPFFzbXtsX1wLCsuAAbkA8JhXKApxvEYmaEd3fpNC";
+          const publicKeyHash = await ledgerSigner.publicKeyHash();
+          const publicKey = await ledgerSigner.publicKey();
+
+          const { rpc } = tezos;
+          const block = await rpc.getBlock();
+          const sourceData = await rpc.getContract(freshAddress);
 
           o.next({ type: "device-signature-requested" });
 
-          let type: OperationType = "OUT";
+          let type: OperationType = "NONE";
 
-          let res, opbytes;
-          const params = {
-            fee: transaction.fees?.toNumber() || 0,
-            storageLimit: transaction.storageLimit?.toNumber() || 0,
-            gasLimit: transaction.gasLimit?.toNumber() || 0,
+          let forgedBytes: string;
+
+          const transactionFees = {
+            fee: (transaction.fees || 0).toString(),
+            gas_limit: (transaction.gasLimit || 0).toString(),
+            storage_limit: (transaction.storageLimit || 0).toString(),
           };
 
+          const contents: OperationContents[] = [];
+
+          if (!(account as TezosAccount).tezosResources.revealed) {
+            const revealFees = await tezos.estimate.reveal();
+
+            contents.push({
+              kind: OpKind.REVEAL,
+              fee: DEFAULT_FEE.REVEAL.toString(),
+              gas_limit: (revealFees?.gasLimit || 0).toString(),
+              storage_limit: (revealFees?.storageLimit || 0).toString(),
+              source: publicKeyHash,
+              counter: (Number(sourceData.counter) + 1).toString(),
+              public_key: publicKey,
+            });
+          }
+
           switch (transaction.mode) {
-            case "send":
-              res = await tezos.contract.transfer({
-                mutez: true,
-                to: transaction.recipient,
-                amount: transaction.amount.toNumber(),
-                ...params,
+            case "send": {
+              type = "OUT";
+
+              contents.push({
+                kind: OpKind.TRANSACTION,
+                amount: transaction.amount.toString(),
+                destination: transaction.recipient,
+                source: freshAddress,
+                counter: (Number(sourceData.counter) + 1 + contents.length).toString(),
+                ...transactionFees,
+              });
+
+              forgedBytes = await rpc.forgeOperations({
```

Comment:
Minor but we could unit test what is provided to rpc.forgeOperations for each operation type